codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/AwsServiceModel.java

```diff
@@ -366,6 +372,15 @@ private MethodSpec getValueForField() {
         return methodBuilder.build();
     }
 
+    private MethodSpec endpointDiscoveryRequest() {
```

Comment:
Do we need to put this on the service model? Seems like an odd place for it.

Reply:
Any alternative suggestions?

If not on the generated models for the requests, we'd have to either generate a separate class per request or generate a list of which fields need to be pulled out for the identifiers for each request.